app/main.py

```diff
@@ -2,5 +1,17 @@
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    func_cache = {}
+
+    def wrapper(*args) -> Any:
+        if args in func_cache:
+            print("Getting from cache")
+            return func_cache[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            func_cache[args] = result
```

Comment:
Ensure that the arguments used as keys in the dictionary are immutable. If the function can receive mutable arguments, consider converting them to a hashable type before storing them in the cache.